.github/workflows/integration.yml

```diff
@@ -136,8 +150,21 @@ jobs:
   e2e-test-trigger:
      runs-on: ubuntu-latest
      needs: [build, configure]
-     if: github.event.client_payload.slash_command.command == 'test'
      steps:
+
+       # Since we are using a repository-dispatch event, we have to explicitly set a run check. We initialize it to a "pending" state.
+       - uses: octokit/request-action@v2.x
+         name: "Initialize run check to 'pending (For PR-only)"
+         with:
+           route: POST /repos/${{ github.repository }}/statuses/${{ needs.configure.outputs.commit-ref }}
+           state: "${{ needs.build.result }}"
+           description: "Component build status"
+           context: "Components building"
+           target_url: "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+         env:
+           GITHUB_TOKEN: ${{ secrets.CI_TOKEN }}
+         if: always() && ${{ needs.build.result != 'cancelled' }}
```

Comment:
Why do we need these `always`?

Reply:
Because they should be executed even if the pipeline has failed (it won't without `always()`).